core/translate/select.rs

```diff
@@ -21,18 +22,25 @@ pub fn translate_select(
     select: ast::Select,
     syms: &SymbolTable,
 ) -> Result<ProgramBuilder> {
-    let mut select_plan = prepare_select_plan(schema, select, syms, None)?;
+    let mut select_plan = prepare_select_plan(schema, select.clone(), syms, None)?;
     optimize_plan(&mut select_plan, schema)?;
-    let Plan::Select(ref select) = select_plan else {
+    let Plan::Select(ref select_ref) = select_plan else {
         panic!("select_plan is not a SelectPlan");
     };
 
     let mut program = ProgramBuilder::new(ProgramBuilderOpts {
         query_mode,
-        num_cursors: count_plan_required_cursors(select),
-        approx_num_insns: estimate_num_instructions(select),
-        approx_num_labels: estimate_num_labels(select),
+        num_cursors: count_plan_required_cursors(select_ref),
+        approx_num_insns: estimate_num_instructions(select_ref),
+        approx_num_labels: estimate_num_labels(select_ref),
     });
+
+    if let ast::OneSelect::Values(values) = &select.body.select.as_ref() {
```

Comment:
I feel like we shouldn't need separate specialcasing to handle this. What about e.g. `SELECT * from sometable JOIN (VALUES (1),(2),(3))?`

It seems `VALUES` with multiple rows is quite similar to how subqueries work.